src/ripple/shamap/SHAMapItem.h

```diff
@@ -149,36 +149,42 @@ intrusive_ptr_release(SHAMapItem const* x)
         if constexpr (!std::is_trivially_destructible_v<SHAMapItem>)
             std::destroy_at(x);
 
-        // If the slabber doens't claim this pointer, it was allocated
+        // If the slabber doesn't claim this pointer, it was allocated
         // manually, so we free it manually.
         if (!detail::slabber.deallocate(const_cast<std::uint8_t*>(p)))
             delete[] p;
     }
 }
 
-inline boost::intrusive_ptr<SHAMapItem>
+inline boost::intrusive_ptr<SHAMapItem const>
 make_shamapitem(uint256 const& tag, Slice data)
 {
     assert(data.size() <= megabytes<std::size_t>(16));
 
     std::uint8_t* raw = detail::slabber.allocate(data.size());
 
-    // If we can't grab memory from the slab allocators, we fall back to
-    // the standard library and try to grab a precisely-sized memory block:
+    // If we can't grab memory from the slab allocators, we use new to
+    // get a precisely-sized and properly aligned memory block:
     if (raw == nullptr)
-        raw = new std::uint8_t[sizeof(SHAMapItem) + data.size()];
+        raw = new (std::align_val_t(alignof(SHAMapItem)))
+            std::uint8_t[sizeof(SHAMapItem) + data.size()];
+
+    assert(reinterpret_cast<std::uintptr_t>(raw) % alignof(SHAMapItem) == 0);
```

Comment:
Why do you need `raw` to be compatibly aligned with `std::uintptr_t` ? I don't see `raw` being type-casted and used as `std::uintptr_t`

Reply:
That's not what's happening here: I am checking whether the pointer to the newly allocated memory is properly aligned for a `SHAMapItem`. It should always be (the code is carefully implemented to ensure that) but I want to be _extra_ careful, because of the placement `new` operation below, which uses [`std::assume_aligned`](https://en.cppreference.com/w/cpp/memory/assume_aligned).

Reply:
Okay, I get that. But why do you choose to type-cast `raw` into `std::uintptr_t` ? 

Would it make a difference if you used `std::intptr_t` (or another pointer type) instead of `std::uintptr_t` ?

Reply:
okay, this makes sense 👍